Kernel/Arch/aarch64/RPi/UART.h

```diff
@@ -24,8 +24,13 @@ class UART {
 
     void print_str(char const* s, size_t length)
     {
-        for (size_t i = 0; i < length; ++i)
-            send(*s++);
+        for (size_t i = 0; i < length; ++i) {
+            char character = *s++;
+            if (character == '\n' && !m_seen_cr)
+                send('\r');
+            send(character);
+            m_seen_cr = character == '\r';
```

Comment:
This logic can be skipped entirely, can't it? Sending `\r` twice in succession shouldn't do anything significant.

Reply:
Yeah, I guess that wouldn't hurt either. I basically copied the logic from here: https://github.com/SerenityOS/serenity/blob/64b29eb4596976138b2e4f67f8502419e584f1e4/Kernel/kprintf.cpp#L41-L70

Reply:
Removed the extra logic :^)